src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -1261,10 +1261,56 @@ bool IFloatingPoint<Half>.TryWriteSignificandLittleEndian(Span<byte> destination
         public static Half Atan2Pi(Half y, Half x) => (Half)float.Atan2Pi((float)y, (float)x);
 
         /// <inheritdoc cref="IFloatingPointIeee754{TSelf}.BitDecrement(TSelf)" />
-        public static Half BitDecrement(Half x) => (Half)MathF.BitDecrement((float)x);
+        public static Half BitDecrement(Half x)
+        {
+            ushort bits = x._value;
+
+            if ((bits & PositiveInfinityBits) >= PositiveInfinityBits)
```

Comment:
In what case is `(bits & PositiveInfinityBits) > PositiveInfinityBits`?

Reply:
As per the comments right below, it covers `NaN`, `PositiveInfinity`, and `NegativeInfinity`.

IEEE 754 floating-point values are one's complement so `PositiveInfinity` and NegativeInfinity` only differ by whether the most significant bit (sign bit) is set.

Bitwise values greater than `PositiveInfinity` represent `+NaN`. Bitwise values less than `NegativeInfinity` represent `-NaN`.

So doing a bitwise-and with `PositiveInfinityBits` strips the sign bit (getting the absolute value) and then ensures that the value is only `infinity or nan`.

Reply:
Ah, yeah, potentially it would be fine with `== PositiveInfinityBits`, but `>=` is what we're already doing in float/double, so its likely best to keep it consistent.